packages/react-native/scripts/codegen/generate-artifacts-executor/utils.js

```diff
@@ -114,8 +139,14 @@ function readReactNativeConfig(projectRoot /*: string */) /*: $FlowFixMe */ {
 function findCodegenEnabledLibraries(
   pkgJson /*: $FlowFixMe */,
   projectRoot /*: string */,
+  baseOutputPath /*: string */,
   reactNativeConfig /*: $FlowFixMe */,
 ) /*: Array<$FlowFixMe> */ {
+  if (!!readGeneratedReactNativeConfig(baseOutputPath)) {
+    // If we ran autolinking, we shouldn't try to run our own "autolinking-like"
+    // library discovery
+    return findLibrariesFromReactNativeConfig(projectRoot, reactNativeConfig);
+  }
```

Comment:
I don't think this will work fine. If an app has some local modules, or defines some modules/components as RNTester does, that code won't be generated. 

The findLibrariesFromReactNativeConfig is executed anyway below. I think we can remove this if, tbf. `findLibrariesFromReactNativeConfig` code will be executed anyway.

Reply:
I can test this against `RNTester` and see if it reproduces there 👍 Main concern here was the other function that does the "second" autolinking.

fwiw the concern I had with the "find libraries" function is that it's either already the responsibility of the input or duplicated here rather (other we'd be missing the pod spec for local modules)

Running this twice could be risky since both results may contend. But I can test it both ways in `RNTester` and report back here

Reply:
Ideally, autolinking should be able to detect local modules as well, otherwise linking might break after codegen and not-include this module. 

But, if it makes everyone a bit more confident, we could also combine the autolinking results, with the result from the shallow discovery? That way, we still ensure all autolinking pages are codegen'd that needs to be codegen'd, we just can't ensure that all codegen'd packages are also linked properly.

Reply:
I think I get what's going on and we were talking about different things. Added the commit to put this back, but just to note, `pkgJson.codegenConfig` and the other logic in here seems a bit risky and "special" to this repo, so I'd be careful about shipping this unflagged in a future release and remove it. Although that said, if we change away from Cocoapods, I suppose this might get a rewrite anyway